### ISSUE TITLE
analyze/optimize routes in teleop

### ISSUE THREAD

**Issue by rivques** [chsbacon/FTC-CENTERSTAGE-7080#52]
in theory we'd like to spend every second of the game moving, either going towards the wing for pixels or going towards the backboard to score. when i was rewatching some of our matches i noticed we were spending a long time not moving because we had to wait for the arm to either get ready to pass under the truss, get into scoring position, or get into intake position. while #50 and #51 will help with this, it's also important which route we take from wing to backboard, which in practice means choosing the correct truss to go under.

here's my thoughts:
first of all: number the passageways according to this image and assume we're on blue (red is just the mirror of this): 

![Image of the FTC Centerstage field, with the columns numbered from 1-6 left-to-right from the audience's perspecitve](https://github.com/chsbacon/FTC-CENTERSTAGE-7080/assets/38469076/9dbf0d7d-7c10-4027-b799-5cec74d48052)

most of the arm motion is done in the back half of the field (only going from truss position to intake position in the front, which is a relatively short move). this means we should spend as much time as possible in the back half to maximise the time our arm has to move around on its longer moves. because of this, passage 1 is the least optimal, because it gives the shortest amount of time in the back half AND it is very easy to run into the other alliance's wing and get a major penalty. passage 2 is only slightly better. passages 3 and 4 (i.e. the stage door) can work well, if you fit (or are going front-to-back) because there's no pillar between 3 and 4 like there is in all other gaps, so you can line yourself up with much less tolerance. if the arm is reasonably fast, this is the best choice. if the arm is slower (or you need to go back-to-front and don't fit under the stage door), then 5 or 6 is best, and imo they're fairly equal and should be picked based on where other robots are and if there is a more optimal angle for intake from the wing.

tldr: 1 and 2 are bad, 3 and 4 (stage door) are best if you can fit and your arm is fast, and 5 and 6 are best if you can't do 3/4.

**Comment by rivques** [comment]
this is kinda hard to practice at home on the half field unfortunately, but it's still good to keep in mind

**Comment by rivques** [comment]
cant do driver practice till we're back
